api/maven-api-core/src/main/java/org/apache/maven/api/services/ChecksumAlgorithmService.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.api.services;
+
+import java.io.*;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.Map;
+
+import org.apache.maven.api.Service;
+import org.apache.maven.api.annotations.Experimental;
+import org.apache.maven.api.annotations.Nonnull;
+
+/**
+ * Checksum algorithms service.
+ *
+ * @since 4.0.0
+ */
+@Experimental
+public interface ChecksumAlgorithmService extends Service {
+    /**
+     * Returns {@link ChecksumAlgorithm} for given algorithm name, or throws if algorithm not supported.
+     *
+     * @throws IllegalArgumentException if asked algorithm name is not supported.
+     */
+    @Nonnull
+    ChecksumAlgorithm select(@Nonnull String algorithmName);
+
+    /**
+     * Returns a list of {@link ChecksumAlgorithm} in same order as algorithm names are ordered, or throws if any of the
+     * algorithm name is not supported. The returned list has equal count of elements as passed in collection of names,
+     * and if names contains duplicated elements, the returned list of algorithms will have duplicates as well.
+     *
+     * @throws IllegalArgumentException if any asked algorithm name is not supported.
+     * @throws NullPointerException if passed in list of names is {@code null}.
+     */
+    @Nonnull
+    Collection<ChecksumAlgorithm> selectList(@Nonnull Collection<String> algorithmNames);
```

Comment:
the doc and name should be adjusted to the list -> collection change in the return type